src/contracts/misc/GhoStewardV2.sol

```diff
@@ -0,0 +1,306 @@
+    require(
+      _isChangePositiveAndIncreaseLowerThanMax(
+        currentExposureCap,
+        newExposureCap,
+        currentExposureCap
+      ),
+      'INVALID_EXPOSURE_CAP_UPDATE'
+    );
+    _gsmTimelocksByAddress[gsm].gsmExposureCapLastUpdated = uint40(block.timestamp);
+    IGsm(gsm).updateExposureCap(newExposureCap);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateGsmBucketCapacity(
+    address gsm,
+    uint128 newBucketCapacity
+  )
+    external
+    onlyRiskCouncil
+    notLocked(_gsmTimelocksByAddress[gsm].gsmBucketCapacityLastUpdated)
+    onlyApprovedGsm(gsm)
+  {
+    (uint256 currentBucketCapacity, ) = IGhoToken(GHO_TOKEN).getFacilitatorBucket(gsm);
+    require(
+      _isChangePositiveAndIncreaseLowerThanMax(
+        currentBucketCapacity,
+        newBucketCapacity,
+        currentBucketCapacity
+      ),
+      'INVALID_BUCKET_CAPACITY_UPDATE'
+    );
+    _gsmTimelocksByAddress[gsm].gsmBucketCapacityLastUpdated = uint40(block.timestamp);
+    IGhoToken(GHO_TOKEN).setFacilitatorBucketCapacity(gsm, newBucketCapacity);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function updateGsmFeeStrategy(
+    address gsm,
+    uint256 buyFee,
+    uint256 sellFee
+  )
+    external
+    onlyRiskCouncil
+    notLocked(_gsmTimelocksByAddress[gsm].gsmFeeStrategyLastUpdated)
+    onlyApprovedGsm(gsm)
+  {
+    address currentFeeStrategy = IGsm(gsm).getFeeStrategy();
+    require(currentFeeStrategy != address(0), 'GSM_FEE_STRATEGY_NOT_FOUND');
+    uint256 currentBuyFee = IGsmFeeStrategy(gsm).getBuyFee(10e5);
+    uint256 currentSellFee = IGsmFeeStrategy(gsm).getSellFee(10e5);
+    require(
+      _isChangePositiveAndIncreaseLowerThanMax(currentBuyFee, buyFee, GSM_FEE_RATE_CHANGE_MAX) &&
+        _isChangePositiveAndIncreaseLowerThanMax(currentSellFee, sellFee, GSM_FEE_RATE_CHANGE_MAX),
+      'INVALID_FEE_STRATEGY_UPDATE'
+    );
+    address cachedStrategyAddress = _gsmFeeStrategiesByRates[buyFee][sellFee];
+    if (cachedStrategyAddress == address(0)) {
+      FixedFeeStrategy newRateStrategy = new FixedFeeStrategy(buyFee, sellFee);
+      cachedStrategyAddress = address(newRateStrategy);
+      _gsmFeeStrategiesByRates[buyFee][sellFee] = cachedStrategyAddress;
+      _gsmFeeStrategies.add(cachedStrategyAddress);
+    }
+    _gsmTimelocksByAddress[gsm].gsmFeeStrategyLastUpdated = uint40(block.timestamp);
+    IGsm(gsm).updateFeeStrategy(cachedStrategyAddress);
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function addApprovedGsms(address[] memory gsms) external onlyOwner {
+    for (uint256 i = 0; i < gsms.length; i++) {
+      _approvedGsmsByAddress[gsms[i]] = true;
+      _approvedGsms.add(gsms[i]);
+    }
+  }
+
+  /// @inheritdoc IGhoStewardV2
+  function removeApprovedGsms(address[] memory gsms) external onlyOwner {
+    for (uint256 i = 0; i < gsms.length; i++) {
+      _approvedGsmsByAddress[gsms[i]] = false;
+      _approvedGsms.remove(gsms[i]);
+    }
+  }
```

Comment:
we can simplify with one single function `approveGsm(address[] memory gsmList, bool approve)`

Reply:
maybe `approve` is not the best term. What about `control`?
mapping can be `controlledGsms` and function `controlGsm` or so